Data/Primitive/UnliftedArray.hs

```diff
@@ -354,14 +396,21 @@ thawUnliftedArray src off len = do
   return dst
 {-# inline thawUnliftedArray #-}
 
+-- | Execute a stateful computation and freeze the resulting array.
+runUnliftedArray
+  :: (forall s. ST s (MutableUnliftedArray s a))
+  -> UnliftedArray a
+runUnliftedArray m = runST $ m >>= unsafeFreezeUnliftedArray
+{-# inline runUnliftedArray #-}
```

Comment:
That's not going to unbox across the `runST`. You should be able to just copy what I did for `Array`....

Reply:
Done.